lib/keycloak/fake_client.rb

```diff
@@ -1,8 +1,10 @@
 module Keycloak
   class FakeClient
-    def initialize
+    def initialize(id: 'foo-bar-id')
```

Comment:
Why FakeClient gets an id? Shouldn't it be an id generator?
Something like:
```
  def initialize(id: ->(email) { ...  })
  ...
  ```
  
  That will allow to have different ids for different users.

Reply:
The reason for allowing an ID to be set is because said ID [is verified in the test.](https://github.com/growthtribeacademy/pupilfirst/blob/store_sso_id/spec/lib/keycloak/setup_student_account_spec.rb)

Reply:
I know - but instead of value you shall get an ID generator - and you should be able to generate different IDs for each created user. `SecureRandom.uuid` does that but will be PITA i tests when you will want to have know IDs.

Reply:
Yeah, I made it so that I can also mock the ID. It'll only resort to SecureRandom.uuid when nothing is passed.